packages/provider/index.js

```diff
@@ -33,33 +34,38 @@ module.exports = {
   },
 
   testConnection: function(options) {
+    const { networks, network } = options;
+    let networkCheckTimeout, networkType;
+    if (networks[network]) {
+      networkCheckTimeout =
+        networks[network].networkCheckTimeout || DEFAULT_NETWORK_CHECK_TIMEOUT;
+      networkType = networks[network].type;
+    } else {
+      networkCheckTimeout = DEFAULT_NETWORK_CHECK_TIMEOUT;
+    }
     const provider = this.getProvider(options);
     const interfaceAdapter = new InterfaceAdapter();
-    const web3 = new Web3Shim({ provider });
+    const web3 = new Web3Shim({ provider, networkType });
```

Comment:
Oh, for when you resolve the merge conflict: 
```suggestion
    const interfaceAdapter = new InterfaceAdapter({ provider, networkType });
    const web3 = new Web3Shim({ provider, networkType });
```

Reply:
Should I add this now?

Reply:
Woops, guess you can't do multi-line suggestions. You can take out the extra var instantiation in the merge conflict.